index.js

```diff
@@ -17,14 +16,21 @@ export default postcss.plugin('postcss-custom-media', opts => {
 	// promise any custom media are imported
 	const customMediaPromise = getCustomMediaFromImports(importFrom);
 
-	return async root => {
-		const customMedia = Object.assign(
-			await customMediaPromise,
-			getCustomMediaFromRoot(root, { preserve })
-		);
+	return {
+		postcssPlugin: 'postcss-custom-media',
+		Root: async root => {
```

Comment:
[Migration guide](https://evilmartians.com/chronicles/postcss-8-plugin-migration#step-2-use-the-updated-api) was updated to use `Once` instead of `Root`. You need PostCSS 8.1+ for this.

```suggestion
		Once: async root => {
```

Reply:
Updated